src/main/java/org/jabref/cli/XMPUtilMain.java

```diff
@@ -92,63 +90,62 @@ public static void main(String[] args) throws IOException, TransformerException
 
                     if (entries.isEmpty()) {
                         System.err.println("Could not find BibEntry in " + args[0]);
-                    } else {
-                        System.out.println(XMPUtil.toXMP(entries, result.getDatabase(), xmpPreferences));
                     }
                 }
             } else {
                 usage();
             }
-            break;
-        case 2:
+        } else if (argsLength == 2) {
             if ("-x".equals(args[0]) && args[1].endsWith(".pdf")) {
                 // Read from pdf and write as BibTex
                 Optional<XMPMetadata> meta = XMPUtil.readRawXMP(new File(args[1]));
 
                 if (meta.isPresent()) {
-                    XMLUtil.save(meta.get().getXMPDocument(), System.out, StandardCharsets.UTF_8.name());
+                    XmpSerializer serializer = new XmpSerializer();
+                    serializer.serialize(meta.get(), System.out, true);
                 } else {
                     System.err.println("The given pdf does not contain any XMP-metadata.");
                 }
-                break;
+                return;
             }
 
             if (args[0].endsWith(".bib") && args[1].endsWith(".pdf")) {
-                ParserResult result = new BibtexParser(importFormatPreferences, Globals.getFileUpdateMonitor()).parse(new FileReader(args[0]));
+                try (FileReader reader = new FileReader(args[0])) {
+                    ParserResult result = new BibtexParser(importFormatPreferences, Globals.getFileUpdateMonitor()).parse(reader);
 
-                Collection<BibEntry> entries = result.getDatabase().getEntries();
+                    Collection<BibEntry> entries = result.getDatabase().getEntries();
 
-                if (entries.isEmpty()) {
-                    System.err.println("Could not find BibEntry in " + args[0]);
-                } else {
-                    XMPUtil.writeXMP(new File(args[1]), entries, result.getDatabase(), false, xmpPreferences);
-                    System.out.println("XMP written.");
+                    if (entries.isEmpty()) {
+                        System.err.println("Could not find BibEntry in " + args[0]);
+                    } else {
+                        XMPUtil.writeXMP(new File(args[1]), entries, result.getDatabase(), false, xmpPreferences);
```

Comment:
shouldn't this also be either a Path?

Reply:
No. The parameter is a file or sometimes also the filePath as a string is possible.